mcdemoaux/vision/vis.py

```diff
@@ -25,3 +25,47 @@ def run(self):
                 cv2.imshow(data[0], data[1])
             cv2.waitKey(300)
 
+class VisualizerMac():
+    def __init__(self):
+        self.queue = deque(maxlen=10)
+
+    def __call__(self, *args):
+        cv2.imshow(args[0], args[1])
+        cv2.waitKey(1)
```

Comment:
With 1 ms here, the visualizer is called too often and eats all processor resources. With much large delay, the agent misses events more often. But maybe a better compromise is possible. Maybe, 3 or 5 ms? Have you checked the behavior of achiever with different delays?